azkaban-execserver/src/main/java/azkaban/execapp/event/JobCallbackManager.java

```diff
@@ -0,0 +1,272 @@
+      }
+    }
+
+    // for completed status
+    List<HttpRequestBase> httpRequestsForCompletedStatus =
+        JobCallbackUtil.parseJobCallbackProperties(props, COMPLETED,
+            contextInfo, maxNumCallBack, jobLogger);
+
+    // now make the call
+    if (!httpRequestsForCompletedStatus.isEmpty()) {
+      jobLogger.info("Making " + httpRequestsForCompletedStatus.size()
+          + " job callbacks for status: " + COMPLETED);
+
+      addDefaultHeaders(httpRequestsForCompletedStatus);
+      JobCallbackRequestMaker.getInstance().makeHttpRequest(jobId, jobLogger,
+          httpRequestsForCompletedStatus);
+    } else {
+      jobLogger.info("No job callbacks for status: " + COMPLETED);
+    }
+  }
+
+  private void processJobCallOnStart(Event event) {
+    JobRunner jobRunner = (JobRunner) event.getRunner();
+
+    if (JobCallbackUtil.isThereJobCallbackProperty(jobRunner.getProps(),
+        JobCallbackStatusEnum.STARTED)) {
+
+      // don't want to waste time resolving properties if there are
+      // callback properties to parse
+      Props props = PropsUtils.resolveProps(jobRunner.getProps());
+
+      Map<String, String> contextInfo =
+          JobCallbackUtil.buildJobContextInfoMap(event, azkabanHostName);
+
+      List<HttpRequestBase> jobCallbackHttpRequests =
+          JobCallbackUtil.parseJobCallbackProperties(props, STARTED,
+              contextInfo, maxNumCallBack, jobRunner.getLogger());
+
+      String jobId = contextInfo.get(CONTEXT_JOB_TOKEN);
+      String msg =
+          String.format("Making %d job callbacks for job %s for jobStatus: %s",
+              jobCallbackHttpRequests.size(), jobId, STARTED.name());
+
+      jobRunner.getLogger().info(msg);
+
+      addDefaultHeaders(jobCallbackHttpRequests);
+
+      JobCallbackRequestMaker.getInstance().makeHttpRequest(jobId,
+          jobRunner.getLogger(), jobCallbackHttpRequests);
+    }
+  }
+
+  private String getAzkabanHostName(Props props) {
+    String baseURL = props.get(JobRunner.AZKABAN_WEBSERVER_URL);
+    try {
+      String hostName = InetAddress.getLocalHost().getHostName();
+      if (baseURL != null) {
+        URL url = new URL(baseURL);
+        hostName = url.getHost() + ":" + url.getPort();
+      }
+      return hostName;
+    } catch (Exception e) {
+      throw new IllegalStateException(
+          "Encountered while getting azkaban host name", e);
+    }
+  }
+
+  private void addDefaultHeaders(List<HttpRequestBase> httpRequests) {
+    if (httpRequests == null) {
+      return;
+    }
+
+    SimpleDateFormat format =
+        new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss z");
+    format.setTimeZone(TimeZone.getTimeZone("GMT"));
+
+    for (HttpRequestBase httpRequest : httpRequests) {
+      httpRequest.addHeader(new BasicHeader("Date", gmtDateFormatter
+          .format(new Date())));
+      httpRequest.addHeader(new BasicHeader("Host", azkabanHostName));
```

Comment:
Why does the DefaultHeader add host azkabanHostName?

For a HTTP request's headers,the Host's value should be the remote server's host rather than the caller